src/Components/Auctions/RailSlider/Arrow.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { SFC } from "react"
+import styled from "styled-components"
+import Icon from "../../Icon"
+
+interface Props {
+  direction: "left" | "right"
+  onClick?: () => void
+}
+
+export const Arrow: SFC<Props> = ({ direction, onClick }) => {
```

Comment:
Out of curiosity

a) why dont you have these things as default exports?
b) why do the pattern of `const Name = () => {}` instead of `function Name() {}`

Reply:
Good questions!

a) We’ve been moving away from default exports for various reasons, this page summarises all of them https://basarat.gitbooks.io/typescript/docs/tips/defaultIsBad.html
b) We use it to be able to cast the function appropriately, afaik if you use the `function` keyword you can only type the individual parameters and return value

Reply:
Whats the reason for not doing `function Arrow({direction, onClick}: Props) {}`?

Reply:
Oh yeah, I hadn’t considered that to be good enough, so no reason. What type of tooling are you referring to?

Reply:
Mostly symbol identification in editors (for jump to symbol) and stuff like git lens.

Reply:
Nice, didn’t know about git lens!

@damassi Do you have any thoughts on function definition style?

Reply:
In terms of information that is available to the compiler, I think function literals are less precise:

<img width="411" alt="screen shot 2018-01-24 at 12 40 39 pm" src="https://user-images.githubusercontent.com/236943/35356050-1fad0436-0104-11e8-8711-05b030305471.png">

Compared to 

<img width="503" alt="screen shot 2018-01-24 at 12 44 15 pm" src="https://user-images.githubusercontent.com/236943/35356108-55a51308-0104-11e8-8d88-4fbe6d4fcd2d.png">

(JSX.Element vs React.StatelessComponent)

That said, I'm personally in the function literal camp as I find it more pleasing to read and less noisy, as well as providing sufficient safety and intellisense assuming the argument is typed. I also like that function literals are hoisted to the top before execution so that a function declared at the bottom of the file can be used in another function located in the middle without erring out; `const Foo = () =>`-style declarations have to be used in the order in which they are declared and do not have free access to the entire scope.

Reply:
Heh some people would say the lack of hoisting is a feature. It is worth saying that `SFC<P>` provides more type safety. if you look at the definition there's much more to it. The question is how these functions are passed around, usually they're used 100% w/in JSX so it doesn't actually matter too much.

This is the ultimate bike shed. My personal opinion is let anyone do it either way, whatever they individually like more.

Reply:
> This is the ultimate bike shed

😄

Reply:
FWIW: In react native, until _very_ recently a function literal `const component = () => <View/>` would not get hot module reloaded. So that's been avoided (or [hijacked](https://github.com/artsy/emission/blob/e106feabfbf4663023e0c7010177527841884b23/src/lib/Components/Consignments/Components/ArtworkConsignmentTodo.tsx#L124-L126)) from the mobile side.